app/src/main/java/de/tum/in/tumcampusapp/models/managers/SurveyManager.java

```diff
@@ -340,12 +327,35 @@ void replaceIntoDBOpenQuestions(Question q) {
     }
 
     void replaceIntoDb(Faculty f) {
-        db.execSQL("REPLACE INTO faculties (faculty, name) VALUES (?, ?)", new String[]{f.getId(), f.getName()});
-    }
+        // Unfortunately I had to do it like that and not with a Replace Into statment because for some reason the replace statement doesn't work correctly
+        Cursor c = db.rawQuery("SELECT * FROM faculties WHERE faculty = ?",new String[]{f.getId()});
+        if(c.moveToFirst()){
```

Comment:
Please optimize this "if". You don't need the try/catch block in both parts of the if.